cmd/bom/cmd/root.go

```diff
@@ -0,0 +1,197 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+	"k8s.io/release/pkg/spdx"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"sigs.k8s.io/release-utils/log"
+)
+
+var rootCmd = &cobra.Command{
+	Short: "bom: A tool for working with SPDX manifests",
+	Long: `bom (Bill of Materials)
+
+bom is a little utility that lets software authors generate
+SPDX manifests to describe the contents of a release. The
+SPDX manifests provide a way to list and verify all items
+contained in packages, images, and individual files while
+packing the data along with licensing information.
+
+bom is still in its early stages and it is an effort to open
+the libraries developed for the Kubernetes SBOM for other 
+projects to use.
+
+`,
+	Use:          "bom",
+	SilenceUsage: false,
+	// SilenceErrors:     true,
+	PersistentPreRunE: initLogging,
+}
+
+var generateCmd = &cobra.Command{
+	Short: "bom generate → Create SPDX manifests",
+	Long: `bom → Create SPDX manifests
+
+generate is the bom subcommand to generate SPDX manifests.
+Currently supports creating SBOM for files, images, and docker
+archives (images in tarballs). Supports remote pulling from
+registries.
+
+`,
+	Use:               "generate",
+	SilenceUsage:      true,
+	SilenceErrors:     true,
+	PersistentPreRunE: initLogging,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		return generateBOM(genOpts)
+	},
+}
+
+type commandLineOptions struct {
+	logLevel string
+}
+
+type generateOptions struct {
+	analyze    bool
+	namespace  string
+	outputFile string
+	images     []string
+	tarballs   []string
+	files      []string
+}
+
+// Validate verify options consistency
+func (opts *generateOptions) Validate() error {
+	if len(opts.images) == 0 && len(opts.files) == 0 && len(opts.tarballs) == 0 {
+		return errors.New("to generate a SPDX BOM you have to provide at least one image or file")
+	}
+	return nil
+}
+
+var (
+	commandLineOpts = &commandLineOptions{}
+	genOpts         = &generateOptions{}
+)
+
+func init() {
+	generateCmd.PersistentFlags().StringSliceVarP(
+		&genOpts.images,
+		"image",
+		"i",
+		[]string{},
+		"list of images",
+	)
+	generateCmd.PersistentFlags().StringSliceVarP(
+		&genOpts.files,
+		"file",
+		"f",
+		[]string{},
+		"list of files to include",
+	)
+
+	generateCmd.PersistentFlags().StringSliceVarP(
+		&genOpts.tarballs,
+		"tarball",
+		"t",
+		[]string{},
+		"list of docker archive tarballs to include in the manifest",
+	)
+
+	generateCmd.PersistentFlags().StringVarP(
+		&genOpts.namespace,
+		"namespace",
+		"n",
+		"",
+		"an URI that servers as namespace for the SPDX doc",
+	)
+
+	generateCmd.PersistentFlags().StringVarP(
+		&genOpts.outputFile,
+		"output",
+		"o",
+		"",
+		"path to the file where the document will be written (defaults to STDOUT)",
+	)
```

Comment:
What's our _preferred_ default behavior here?
Given we're producing documents, we may want to always write something to file somewhere, unless explicitly told otherwise?

I don't have strong opinions here yet, so consider this comment non-blocking.

Reply:
Let's discuss it for 0.0000002. I find it useful to output to STDERR, but this is just me debugging (and we can always have `--output=-` )